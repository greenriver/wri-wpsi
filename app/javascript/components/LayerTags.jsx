import React from 'react'
import { withRouter } from 'react-router-dom'

const CATEGORIES = window.categories

const LayerTags = ({ history, layer, excludedTag }) => {
  const tagStyle = {
    marginRight: '15px',
    fontSize: 'smaller',
    borderRadius: '16px',
    backgroundColor: '#F4F4F4',
    padding: '5px 15px',
    cursor: 'pointer',
  }

  return layer.category_slugs.map(category_slug => (
    category_slug === excludedTag ?
      null :
      <span
        style={tagStyle}
        key={category_slug}
        onClick={() => history.push(`/map/datasets/${category_slug}`)}
      >
        {CATEGORIES.find(category => category.slug == category_slug).title}
      </span>
  ))
}

import PropTypes from 'prop-types'
LayerTags.propTypes = {
  history: PropTypes.object.isRequired,
  layer: PropTypes.object.isRequired,
  excludedTag: PropTypes.string,
}

export default withRouter(LayerTags)
