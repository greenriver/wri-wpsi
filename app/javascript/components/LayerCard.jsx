import React from 'react'
import { Link } from 'react-router-dom'
import LayerTags from 'components/LayerTags'
import styleVariables from 'components/styles/variables'
import injectSheet from 'react-jss'

const styleVars = styleVariables()
const sourceStyle = {
  fontStyle: 'italic',
  fontSize: '13px',
  lineHeight: '1.2'
}
const styles = {
  container: {
    marginBottom: '10px',
    border: `1px solid ${styleVars.colors.gray2}`,
    borderTopRightRadius: 4,
    borderBottomRightRadius: 4,
    background: 'rgba(255,255,255,1)',
    flex: '1 1 auto',
    display: 'flex',
    flexDirection: 'column',
  },
  toggle: {
    marginBottom: 0,
    flex: '0 1 auto',
    boxShadow: 'none',
    borderRadius: 'none',
  },
  white: {
    borderRadius: 4,
    backgroundColor: 'white',
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    lineHeight: '2em',
    padding: '10px 15px 0 15px',
  },
  title: {
    marginBottom: 5,
  },
  content: {
    padding: '0 15px',
    flex: '1 0 auto',
  },
  contentDescription: {
    paddingRight: '10%',
    margin: '0 0 5px 0',
  },
  moreLink: {
    paddingLeft: 5,
    textDecoration: 'none',
    marginLeft: 'auto',
  },
  footer: {
    borderTop: `1px solid ${styleVars.colors.gray3}`,
    padding: '10px 15px',
    display: 'flex',
    marginTop: 'auto',
    flex: '0 1 auto',
  },
  source: {
    margin: '0 0 15px 0',
  },
  sourceLink: sourceStyle,
  sourceDescription: sourceStyle,

}

const LayerCard = ({
  layer,
  variant,
  classes,
  excludedTag,
  secondaryAction,
  children,
}) => {
  const containerClass = `${classes.container} ${variant ? classes[variant] : ''}`
  return (
    <div className={containerClass}>
      <header className={classes.header}>
        <h2 className={classes.title}>{layer.name}</h2>
        {secondaryAction}
      </header>
      <div className={classes.content}>
        <p className={classes.contentDescription}>
          {layer.short_description}
        </p>
        <p className={classes.source}>
          <a className={classes.sourceLink} href={layer.source_url}>{layer.source_name}</a>
          <span className={classes.sourceDescription}>{layer.source_description && `, ${layer.source_description}`}</span>
        </p>
        <div style={{marginBottom: '15px'}}>
          <LayerTags layer={layer} excludedTag={excludedTag} />
        </div>
      </div>
      <footer className={classes.footer}>
        <Link to={`/map/learn_more/${layer.id}`} className={classes.moreLink}>
          <i className='icon__book-reader' style={{marginRight: 5}} />
          <span>Learn more</span>
        </Link>
      </footer>
      {children}
    </div>
  )
}

import PropTypes from 'prop-types'
LayerCard.propTypes = {
  layer: PropTypes.object.isRequired,
  variant: PropTypes.string,
  excludedTag: PropTypes.string,
  secondaryAction: PropTypes.object,
  children: PropTypes.object,
  classes: PropTypes.object,
}

export default injectSheet(styles)(LayerCard)
