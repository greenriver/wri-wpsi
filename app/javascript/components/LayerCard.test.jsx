import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import LayerCard from 'components/LayerCard'
import LayerTags from 'components/LayerTags'
import renderer from 'react-test-renderer'

const layer = {
  id: '123',
  name: 'My test layer',
  category_slugs: ['water', 'peace', 'security'],
  source_name: 'Wikipedia',
  source_url: 'www.wikipedia.org',
}

const component = shallow(<LayerCard layer={layer} />)

it('renders correctly', () => {
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

it('renders a <LayerTags /> component with three tags', () => {
  expect(
    component.containsMatchingElement(
      <LayerTags layer={layer} />
    )
  ).toBeTruthy()
})

it('renders the source name as a link', () => {
  const link = component.find('a')
  expect(link.text()).toEqual('Wikipedia')
  expect(link.props().href).toEqual('www.wikipedia.org')
})
