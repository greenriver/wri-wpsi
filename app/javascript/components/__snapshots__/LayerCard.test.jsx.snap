// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="LayerCard-container-0-1-1 "
>
  <header
    className="LayerCard-header-0-1-4"
  >
    <h2
      className="LayerCard-title-0-1-5"
    >
      My test layer
    </h2>
  </header>
  <div
    className="LayerCard-content-0-1-6"
  >
    <p
      className="LayerCard-contentDescription-0-1-7"
    />
    <p
      className="LayerCard-source-0-1-10"
    >
      <a
        className="LayerCard-sourceLink-0-1-11"
        href="www.wikipedia.org"
      >
        Wikipedia
      </a>
      <span
        className="LayerCard-sourceDescription-0-1-12"
      />
    </p>
    <div
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <withRouter(LayerTags)
        layer={
          Object {
            "category_slugs": Array [
              "water",
              "peace",
              "security",
            ],
            "id": "123",
            "name": "My test layer",
            "source_name": "Wikipedia",
            "source_url": "www.wikipedia.org",
          }
        }
      />
    </div>
  </div>
  <footer
    className="LayerCard-footer-0-1-9"
  >
    <Link
      className="LayerCard-moreLink-0-1-8"
      to="/map/learn_more/123"
    >
      <i
        className="icon__book-reader"
        style={
          Object {
            "marginRight": 5,
          }
        }
      />
      <span>
        Learn more
      </span>
    </Link>
  </footer>
</div>
`;
